src/app/views/PublishesConnections.svelte

```diff
@@ -0,0 +1,144 @@
+          newConnectionStatus["Failed to log in"] = (
+            newConnectionStatus["Failed to log in"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.Error) {
+          newConnectionStatus["Failed to connect"] = (
+            newConnectionStatus["Failed to connect"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.Closed) {
+          newConnectionStatus["Waiting to reconnect"] = (
+            newConnectionStatus["Waiting to reconnect"] || new Map()
+          ).set(url, cxn)
+        } else if (cxn.socket.status === SocketStatus.New) {
+          newConnectionStatus["Not connected"] = (
+            newConnectionStatus["Not connected"] || new Map()
+          ).set(url, cxn)
+        } else if (getRelayQuality(cxn.url) < 0.5) {
+          newConnectionStatus["Unstable connection"] = (
+            newConnectionStatus["Unstable connection"] || new Map()
+          ).set(url, cxn)
+        } else {
+          newConnectionStatus["Connected"] = (newConnectionStatus["Connected"] || new Map()).set(
+            url,
+            cxn,
+          )
+        }
+      }
+      connectionsStatus = newConnectionStatus
+    }, 800)
+
+    return () => {
+      clearInterval(interval)
+    }
+  })
+</script>
+
+<SelectButton {options} bind:value={selectedOptions} multiple class="text-left">
+  <div class="flex items-center gap-2" slot="item" let:option>
+    {Array.from(connectionsStatus[option]?.values() || []).length || 0}
+    {option}
+  </div>
+</SelectButton>
+{#each connections as cxn (cxn.url)}
+  {@const relay = $relaysByUrl.get(cxn.url)}
+  <AltColor
+    background
+    class="cursor-pointer justify-between rounded-md p-6 shadow"
+    on:click={() => {
+      selected = cxn.url
+      activeTab = "notices"
+    }}>
+    <div class="flex min-w-0 shrink-0 items-start gap-3">
+      {#if relay?.profile?.icon}
+        <img class="h-9 w-9 shrink-0 rounded-full border" src={relay.profile.icon} />
+      {:else}
+        <div class="flex h-9 w-9 shrink-0 items-center justify-center rounded-full border">
+          <i class="fa fa-server text-xl text-neutral-100"></i>
+        </div>
+      {/if}
+      <div class="shrink-0">
+        <div class="flex items-center gap-2">
+          <div class="text-md overflow-hidden text-ellipsis whitespace-nowrap">
+            {displayRelayUrl(cxn.url)}
+          </div>
+        </div>
+        <div class="flex gap-4 text-xs text-neutral-400">
+          {#if relay?.profile?.supported_nips}
+            <span>
+              {relay.profile.supported_nips.length} NIPs
+            </span>
+          {/if}
+          <span>
+            Connected {quantify(relay?.stats?.open_count || 0, "time")}
+          </span>
+        </div>
+      </div>
+      <div class="flex w-full items-center justify-end gap-2 text-sm">
+        {#each options as opt}
+          {#if connectionsStatus[opt]?.has(cxn.url)}
```

Comment:
Conditionals inside a loop inside a flex container with `gap` will result in odd spacing. Filter `options` instead.